src/bsc/BscProjectThreaded.ts

```diff
@@ -26,12 +27,38 @@ export class BscProjectThreaded implements ExtractMethods<BscProject> {
         return this.activateDeferred.isCompleted;
     }
 
+    /**
+     * If an error occurs at any time during the worker's lifetime, it will be caught and stored here.
+     */
+    public isErrored() {
+        return this.errorDeferred.isCompleted;
+    }
+
+    private ready() {
+        return Promise.race([
+            //if we've encountered an error, reject immediately. The underlying error should propagate up
+            this.errorDeferred.promise,
+            //wait for the activate to finish. This should typically be the only promise that resolves
+            this.activateDeferred.promise
+        ]);
+    }
+
     public async activate(options: Parameters<ProgramBuilder['run']>[0]) {
         const timeEnd = this.logger.timeStart('log', 'activate');
 
         // start a new worker thread or get an unused existing thread
         this.worker = bscProjectWorkerPool.getWorker();
 
+        //!!!IMPORTANT!!! this observer must be registered in order to prevent the worker thread from crashing the main thread
+        this.worker.on('error', (error) => {
+            this.logger.error('Worker encountered an error', error);
+            this.errorDeferred.reject(error);
+            //!!!IMPORTANT!!! this is required to prevent node from freaking out about an uncaught promise
+            this.errorDeferred.promise.catch(e => {
```

Comment:
This is also important. It's new code that helps track whether our worker has crashed. but by not observing the rejection, that would _also_ bring down the node process. So we observe it, do nothing, and assume someone else will pay attention on the outside.